### PR TITLE
For review: Chapter 1 - Introduction to cloud-native development

### PR DESCRIPTION
**Pull request by GeoffAllix**
Please add review comments on this PR.

Please review for technical content. There will be a copy edit pass before release.

### CODE REVIEW

**Review by GeoffAllix — COMMENTED** [review]
jamesmontemagno2 weeks ago
I wonder if we have a Microsoft example.....

GeoffAllix2 weeks ago
Is this something that you're looking into?

File: docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Not all is bad. Monoliths offer some distinct advantages. For example, they're straightforward to...
+
+- build
+- test
+- deploy
+- troubleshoot
+- vertically scale
+
+Many successful apps that exist today were created as monoliths. The app is a hit and continues to evolve, iteration after iteration, adding more functionality.
```
Comment (alistairmatthews) [inline docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md]: I'd suggest "... as you add more functionality". The app isn't adding it.

File: docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+At some point, however, you begin to feel uncomfortable. You find yourself losing control of the application. As time goes on, the feeling becomes more intense, and you eventually enter a state known as the `Fear Cycle`:
+
+- The app has become so overwhelmingly complicated that no single person understands it.
+- You fear making changes - each change has unintended and costly side effects.
+- New features/fixes become tricky, time-consuming, and expensive to implement.
+- Each release becomes as small as possible and requires a full deployment of the entire application.
+- One unstable component can crash the entire system.
+- New technologies and frameworks aren't an option.
+- It's difficult to implement agile delivery methodologies.
```
Comment (alistairmatthews) [inline docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md:50]: I think "agile" should have a capital.

File: docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Sound familiar?
+
+Many organizations have addressed this monolithic fear cycle by adopting a cloud-native approach to building systems. Figure 1-2 shows the same system built applying cloud-native techniques and practices.
+
+![Cloud-Native Design](./media/cloud-native-design.png)
+
+**Figure 1-2**. Cloud-native design
+
+Note how the application is decomposed across a set of small isolated microservices. Each service is self-contained and encapsulates its own code, data, and dependencies. Each is deployed in a software container and managed by a container orchestrator. Instead of a large relational database, each service owns it own datastore, the type of which vary based upon the data needs. Note how some services depend on a relational database, but other on NoSQL databases. One service stores its state in a distributed cache. Note how all traffic routes through an API Gateway service that is responsible for routing traffic to the core back-end services and enforcing many cross-cutting concerns. Most importantly, the application takes full advantage of the scalability, availability, and resiliency features found in modern cloud platforms.
```
Comment (alistairmatthews) [inline docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/introduction-to-cloud-native-applications.md]: Use a markdown block quote here?

> Like this?

**Comment by GeoffAllix** [comment]
[jamesmontemagno](https://github.com/jamesmontemagno) commented [3 weeks ago](https://github.com/PhilStollery/docs/pull/7#issuecomment-2155530330)
high-level-architecture-diagram

We will probably want to udpate photos as well.

Lots of new ones https://github.com/dotnet/docs/pull/41184 for reference

**Comment by GeoffAllix** [comment]
[jamesmontemagno](https://github.com/jamesmontemagno) commented [16 hours ago](https://github.com/PhilStollery/docs/pull/7#issuecomment-2187546008)
@GeoffAllix we want to use commodities

**Comment by GeoffAllix** [comment]
image

[jamesmontemagno](https://github.com/jamesmontemagno) commented [3 weeks ago](https://github.com/PhilStollery/docs/pull/7#issuecomment-2155530330)
high-level-architecture-diagram

We will probably want to udpate photos as well.

Lots of new ones https://github.com/dotnet/docs/pull/41184 for reference

File: docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/what-is-cloud-native.md [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+> *Factor \#6  specifies "Each microservice should execute in its own process, isolated from other running services."*
+
+### Why microservices?
+
+Microservices provide agility.
+
+Earlier in the chapter, we compared an eCommerce application built as a monolith to that with microservices. In the example, we saw some clear benefits:
+
+- Each microservice has an autonomous lifecycle and can evolve independently and deploy frequently. You don't have to wait for a quarterly release to deploy a new feature or update. You can update a small area of a live application with less risk of disrupting the entire system. The update can be made without a full redeployment of the application.
+
```
Comment (alistairmatthews) [inline docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/what-is-cloud-native.md]: You might want to mention that each microservice is often written by a dediicated and independent team.

File: docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/what-is-cloud-native.md [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+Figure 1-6 shows three different microservices, each in its own container, all running on a single host.
+
+![Multiple containers running on a container host](./media/hosting-mulitple-containers.png)
+
+**Figure 1-6**. Multiple containers running on a container host
+
+Note how each container maintains its own set of dependencies and runtime, which can be different from one another. Here, we see different versions of the Product microservice running on the same host. Each container shares a slice of the underlying host operating system, memory, and processor, but is isolated from one another.
+
+Note how well the container model embraces the **Dependencies** principle from the [Twelve-Factor Application](https://12factor.net/).
+
```
Comment (alistairmatthews) [inline docs/architecture/distributed-cloud-native-apps-containers/introduction-to-cloud-native-development/what-is-cloud-native.md:211]: Hmmm... "Note how..." twice in as many paragraphs?

**Review by alistairmatthews — COMMENTED** [review]
Just a few comments.

**Pull request merged**